examples/ex9.cpp

```diff
@@ -144,6 +144,7 @@ int main(int argc, char *argv[])
    int order = 3;
    bool pa = false;
    bool ea = false;
+   bool fa = false;
```

Comment:
This example doesn't work when I run with `-d cuda` but without any of `-pa`, `-ea`, or `-fa`. Not completely sure if this is related to changes in this PR.

Reply:
The problem doesn't come from this PR.

Reply:
`DSmoother` is not yet GPU ready.

Reply:
Good catch. I suppose we can always use `OperatorJacobiSmoother` (even in the matrix-based case). But that is not so relevant for this PR.

Reply:
We would have to make `OperatorJacobiSmoother` work in the matrix case first though.

Reply:
done.